Dnssec-aware functionality

andreavivaldi opened the pull request.

elgs commented:
Thank you! I need some time to understand the changes in `removeComments` and `flatten`.

andreavivaldi commented:
No problem. Take your time. Let me know if you need some explanation... I'm sorry that there is no comment in the code.
Also I invite you to run some other tests.
Feel free to report me any (probable) error or misunderstanding of the flow.

I can explain right now why I've changed the flatten function:I've rewritten it not to delete all the comments on DNSKEY RR, because they contains the Key ID
